tests/python/unittest/test_codegen_x86.py

```diff
@@ -0,0 +1,59 @@
+import tvm
+import re
+import os
+import ctypes
+
+def test_fp16_to_fp32_with_f16c():
+    target = 'llvm -mcpu=core-avx2 -mattr=+f16c'
+    elements = 64
+    n = tvm.convert(elements)
+    A = tvm.placeholder((n, 8), dtype="float16", name='A')
+    B = tvm.compute(A.shape, lambda *i: A(*i).astype("float32"), name='B')
+    s = tvm.create_schedule(B.op)
+    s[B].vectorize(s[B].op.axis[1])
+    f = tvm.build(s, [A, B], target)
+
+    # Verify we see the correct number of vpaddl and vcnt instructions in the assembly
+    ll = f.get_source('ll')
+    assembly = f.get_source('asm').splitlines()
+    matches = [l for l in assembly if re.search("vcvtph2ps.*ymm", l)]
+    assert (len(matches) > 1)
+
+def test_fp16_to_fp32_with_avx512():
+    target = 'llvm -mcpu=skylake-avx512 -mattr=+avx512f,+f16c'
+    elements = 64
+    n = tvm.convert(elements)
+    A = tvm.placeholder((n, 16), dtype="float16", name='A')
+    B = tvm.compute(A.shape, lambda *i: A(*i).astype("float32"), name='B')
+    s = tvm.create_schedule(B.op)
+    s[B].vectorize(s[B].op.axis[1])
+    f = tvm.build(s, [A, B], target)
+
+    # Verify we see the correct number of vpaddl and vcnt instructions in the assembly
+    ll = f.get_source('ll')
+    assembly = f.get_source('asm').splitlines()
+    matches = [l for l in assembly if re.search("vcvtph2ps.*zmm", l)]
+    assert (len(matches) > 1)
+
+def test_fp16_to_fp32_without_f16c():
+    target = 'llvm'
+    elements = 64
+    n = tvm.convert(elements)
+    A = tvm.placeholder((n, 8), dtype="float16", name='A')
```

Comment:
How is fp16 handled without F16C?

Reply:
Without f16c (i.e. vector/scalar vcvtph2ps), we invoke compiler-rt's builtins i.e. https://github.com/llvm-mirror/compiler-rt/blob/7a739a0dfb6d408c6d587e5c7b52abd89fc3fdd3/lib/builtins/fp_extend_impl.inc#L40.

https://godbolt.org/z/23tsVK is an example.